.github/workflows/ci.yml

```diff
@@ -7,9 +7,35 @@ on:
   pull_request:
     branches:
       - '*'
+permissions:
+  contents: read
 
 jobs:
+  filter:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    outputs:
+      run_tests: ${{ steps.filter.outputs.run_tests }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          # Test related paths are kept in the sorted order top to bottom.
+          filters: |
+            run_tests:
```

Comment:
How will we ensure that every new package gets added to this? Should we instead do something like "!**/.md" etc.?

Reply:
Yeah, this is a problem.

I tried to the same thing but the problem with the !**/*.md is we can only use provide one predicate. If we provide such multiple predicates then it'll not work as intended.

Other simple options is to path-ignore at workflow level and make these tests optional.

Reply:
Recently added feature `predicate-quantifier: 'every'` seems working.